web-app/src/components/BookingModal/index.js

```diff
@@ -1,102 +1,160 @@
 import React from 'react';
-import DatePicker from 'react-datepicker';
 import { PropTypes as PT } from 'prop-types';
-import { Button, Modal } from '../common';
+import { Modal, Form, Input } from '../common';
 import FontAwesomeIcon from '@fortawesome/react-fontawesome';
 import { faTimes } from '@fortawesome/fontawesome-free-solid';
-import { StyleContainer, BookingStatus, BookingInputs, StatusDot, ButtonWrap } from './styled';
+import {
+  StyleContainer,
+  BookingStatus,
+  StatusDot,
+  FormContainer,
+} from './styled';
 import { statusText } from '../../utilities/holidayStatus';
 
 const BookingModal = props => {
   const {
     closeModal,
     showModal,
     booking,
-    changeStart,
-    changeEnd,
-    changeHalfday,
-    requestHoliday,
-    cancelHoliday,
-    updateHoliday,
-    changeWFH,
+    submitForm,
+    formStatus,
+    formIsValid,
   } = props;
 
-  const getActions = () => (
-    booking.title ?
-      <ButtonWrap>
-        <Button id="updateHolidayBtn" onClick={updateHoliday} label={'Update'} />
-        <Button id="cancelHolidayBtn" onClick={cancelHoliday} label={'Cancel Booking'} />
-      </ButtonWrap>
-      :
-      <Button id="requestHolidayBtn" onClick={requestHoliday} label={booking.isWFH ? 'Add' : 'Request'} />
-  );
+  console.log('formIsValid :', formIsValid);
+
+  const isDateDisabled = () => {
+    if (booking.formData.isWFH || booking.formData.isHalfday) {
+      return true;
+    } else if (booking.duration === 0.5) {
+      return false;
+    }
+  };
 
-  const getTotalDays = () => {
-    if (booking.isWFH) { return '0'; }
-    return booking.duration / (booking.isHalfday ? 2 : 1);
+  const getStartDateLabel = () => {
+    if (booking.formData.isWFH || booking.formData.isHalfday) {
+      return 'Date:';
+    } else {
+      return 'Start Date:';
+    }
   };
 
+  let ctas;
+  if (!booking.title) {
+    ctas = [
+      {
+        label: 'Request',
+        event: props.submitForm,
+        disabled: !formIsValid,
+      },
+    ];
+  } else {
+    ctas = [
+      {
+        label: 'Update',
+        event: props.submitForm,
+        disabled: !formIsValid,
+      },
+      {
+        label: 'Cancel',
+        event: props.submitForm,
+        disabled: !formIsValid,
+      },
+    ];
+  }
+
+  const form = (
+    <Form
```

Comment:
These form components take up a lot of lines. Is there no way we could have a subdirectory with forms. Each file imports the form component and then you do what you need to do to it. After that you import it into this component with this one being FormBookingModal. Just something to think about. I think it would be tidier and cleaner to read personally. Abstracting the Forms to their own file. Personal opinion of course you guys discuss it as you're working on it everyday so you may have reasons for this.

Reply:
@paulmatchett This is why I’d mentioned pulling the form out of index and creating a form.jsx file in the same directory.